src/Features/Core/Portable/Diagnostics/AbstractHostDiagnosticUpdateSource.cs

```diff
@@ -1,119 +0,0 @@
-/// which may not be related to any given project/document in the solution.
-/// For example, these include diagnostics generated for exceptions from third party analyzers.
-/// </summary>
-internal abstract class AbstractHostDiagnosticUpdateSource
-{
-    private ImmutableDictionary<DiagnosticAnalyzer, ImmutableHashSet<DiagnosticData>> _analyzerHostDiagnosticsMap =
-        ImmutableDictionary<DiagnosticAnalyzer, ImmutableHashSet<DiagnosticData>>.Empty;
-
-    public abstract Workspace Workspace { get; }
-
-    public event EventHandler<ImmutableArray<DiagnosticsUpdatedArgs>>? DiagnosticsUpdated;
-
-    public void RaiseDiagnosticsUpdated(ImmutableArray<DiagnosticsUpdatedArgs> args)
-    {
-        if (!args.IsEmpty)
-            DiagnosticsUpdated?.Invoke(this, args);
-    }
-
-    public void ClearAnalyzerReferenceDiagnostics(AnalyzerFileReference analyzerReference, string language, ProjectId projectId)
-    {
-        // Perf: if we don't have any diagnostics at all, just return right away; this avoids loading the analyzers
-        // which may have not been loaded if you didn't do too much in your session.
-        if (_analyzerHostDiagnosticsMap.Count == 0)
-            return;
-
-        using var argsBuilder = TemporaryArray<DiagnosticsUpdatedArgs>.Empty;
-        var analyzers = analyzerReference.GetAnalyzers(language);
-        AddArgsToClearAnalyzerDiagnostics(ref argsBuilder.AsRef(), analyzers, projectId);
-        RaiseDiagnosticsUpdated(argsBuilder.ToImmutableAndClear());
-    }
-
-    public void AddArgsToClearAnalyzerDiagnostics(ref TemporaryArray<DiagnosticsUpdatedArgs> builder, ImmutableArray<DiagnosticAnalyzer> analyzers, ProjectId projectId)
-    {
-        foreach (var analyzer in analyzers)
-        {
-            AddArgsToClearAnalyzerDiagnostics(ref builder, analyzer, projectId);
-        }
-    }
-
-    public void AddArgsToClearAnalyzerDiagnostics(ref TemporaryArray<DiagnosticsUpdatedArgs> builder, ProjectId projectId)
-    {
-        foreach (var (analyzer, _) in _analyzerHostDiagnosticsMap)
-        {
-            AddArgsToClearAnalyzerDiagnostics(ref builder, analyzer, projectId);
-        }
-    }
-
-    private void AddArgsToClearAnalyzerDiagnostics(ref TemporaryArray<DiagnosticsUpdatedArgs> builder, DiagnosticAnalyzer analyzer, ProjectId projectId)
-    {
-        if (!_analyzerHostDiagnosticsMap.TryGetValue(analyzer, out var existing))
-        {
-            return;
-        }
-
-        // Check if analyzer is shared by analyzer references from different projects.
-        var sharedAnalyzer = existing.Contains(d => d.ProjectId != null && d.ProjectId != projectId);
-        if (sharedAnalyzer)
-        {
-            var newDiags = existing.Where(d => d.ProjectId != projectId).ToImmutableHashSet();
-            if (newDiags.Count < existing.Count &&
-                ImmutableInterlocked.TryUpdate(ref _analyzerHostDiagnosticsMap, analyzer, newDiags, existing))
```

Comment:
similarly.  nothing was ever reading this _analyzerHostDiagnosticsMap.  removnig that pulled on more of these therads.